agent/agent.go

```diff
@@ -295,10 +295,18 @@ func (a *Agent) prepareNextTask(ctx context.Context, nextTask *apimodels.NextTas
 func nextTaskHasDifferentTaskGroupOrBuild(nextTask *apimodels.NextTaskResponse, tc *taskContext) bool {
 	if tc.taskConfig == nil ||
 		nextTask.TaskGroup == "" ||
-		nextTask.TaskGroup != tc.taskGroup ||
 		nextTask.Build != tc.taskConfig.Task.BuildId {
 		return true
 	}
+	if nextTask.TaskGroup != tc.taskGroup {
+		tc.logger.Task().WarningWhen(nextTask.TaskGroup == tc.taskConfig.Task.TaskGroup, message.Fields{
```

Comment:
Are you certain `logger` will always be non-nil here? It would be safer to wrap it in a nil check.

Reply:
That's fair; I wasn't sure if we needed to since we use it in runPostGroupCommands without error checking, but that might be an oversight